### DISCUSSION TITLE
Creating an Autopilot for Alpha Hold

### DISCUSSION THREAD

**Discussion by beltzjr** [JSBSim-Team/jsbsim#933] [Q&A]
Fairly new to using JSBSim and I am trying to create an autopilot holding a specified AOA. So far everything I try I get zero input to any control surface. I haven't successfully found documentation that I can implement. So far I have added the following.

1. To my aircraft I added: ```<autopilot file="MyAutopilot"/>```
2. Created a file "MyAutopilot.xml" at the same level.

3. Modified my run script trying to set alpha_setpoint.
```xml
<event name="Hold Pitch">
        <condition> position/h-agl-ft > 2 </condition>
        <set name="ap/alpha_setpoint" value="2.0"/>
        <set name="ap/alpha_hold" value="1"/>
      <set name="simulation/output/log_rate_hz" value="10"/>
      <notify>
        <property caption="Cal. airspeed (kts):  "> velocities/vc-kts</property>
        <property caption="Altitude (AGL, ft):   "> position/h-agl-ft</property>
      </notify>
      <notify/>
    </event>
```

My code runs but my elevator-cmd is ALWAYS zero even tho my AOA is not 2. Where am I going wrong?
[MyAutopilot.txt](https://github.com/JSBSim-Team/jsbsim/files/11889015/MyAutopilot.txt)

**Comment by seanmcleod** [comment]
Have you logged all the relevant inputs?

```
aero/alpha-error
aero/ap-alpha-hold-switch
aero/windup-trigger
aero/alpha-hold-pid
```

**Reply by beltzjr** [reply]
Just did and attached the long below. My value in my Autopilot are probobly wrong, but surprised that fcs/elevator-pos-rad and fcs/elevator-cmd-norm are always zero. Am I tracking the wrong variable?
[debugingAlphaHold.txt](https://github.com/JSBSim-Team/jsbsim/files/11897588/debugingAlphaHold.txt)

**Comment by seanmcleod** [comment]
Hmm, is this some sort of spaceship? I see the altitude is over 100,000ft.

Okay, so you're getting a non-zero output from your PID controller.

```sh
Repeating Notify (Event 2) executed at time: 30.008321
    aero/alpha-hold-pid = 0.300012
    fcs/elevator-pos-rad = 0.000000
    fcs/elevator-cmd-norm = 0.000000
```

Which is then having it's sign flipped and output to `ap/elevator_cmd`.

```xml
  <pure_gain name="fcs/elevator">
    <input> aero/alpha-hold-pid </input>
    <gain> -1.0 </gain>
    <output> ap/elevator_cmd </output>
  </pure_gain>
```

But looking at your autopilot code you shared the property `ap/elevator_cmd` isn't connected to the actual elevator, it's a dead-end property.  Unless you've explicitly referenced it in your aircraft's FCS section, however you haven't shared the FCS section so it's impossible to know for sure, whether it's not used anywhere, or you are trying to make use of it to drive the elevator but there is some bug in the way you've connected it.
